syntax = "proto3";

package service_api;

service ThermostatService {
	rpc ControlThermostat(ThermostatApiRequest) returns (ThermostatApiResponse);
}

message ThermostatApiRequest {
	RequestType request_type = 1;
	float set_temperature = 2;
	uint32 time_resolution_ms = 3;
}

enum RequestType {
	START = 0;
	STOP = 1;
	GET_TEMP = 2;
}

message ThermostatApiResponse {
	StatusCode status = 1;
	string message = 2;
}

enum StatusCode {
	SUCCESS = 0;
	FAILURE = 1;
}